.github/workflows/ci.yaml

```
@@ -17,57 +17,62 @@ on:
 jobs:
 
   pre-commit-checks:
-    name: "pre-commit hooks - Python 3.8"
+    name: Pre-commit Checks
     runs-on: ubuntu-latest
     steps:
       - name: Checkout branch
         uses: actions/checkout@v4
-      - name: Install Python 3.8
-        uses: actions/setup-python@v5
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@v0.8.1
         with:
-          python-version: 3.8
-      - name: Run pre-commit checks
-        uses: pre-commit/action@v3.0.1
+          environments: default lint
+      - name: pre-commit
+        run: pixi run pre-commit-run --color=always --show-diff-on-failure
 
   unit-tests:
-    name: "unit tests - Python ${{ matrix.PYTHON_VERSION }} - SA ${{ matrix.SA_VERSION }}"
+    name: "unit tests"
     runs-on: ubuntu-latest
-    env:
-      CI: True
     strategy:
       fail-fast: true
       matrix:
-        PYTHON_VERSION: ['3.8', '3.9', '3.10']
-        SA_VERSION: ['<2.0', '>=2.0']
+        env:
+          - py38
+          - py39
+          - py310
+          - py311
+          - py312
+          - sa1
+          - sa2
```

```suggestion
        env:
          - py38
          - py39
          - py310
          - py311
          - py312
          - sa1
          - sa2
        os:
          - ubuntu-latest
          - macos-latest
          - windows-latest
```

could be interesting as well

---

I added them here:
[1cfafb5](https://github.com/Quantco/datajudge/pull/230/commits/1cfafb5060805f8e006bd22404c95749cf071a73)